clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -752,6 +752,26 @@ public void testFencedMemberStopHeartbeatUntilItReleasesAssignmentToRejoin() {
         assertEquals(1, result.unsentRequests.size(), "Fenced member should resume heartbeat after transitioning to JOINING");
     }
 
+    @ParameterizedTest
+    @ApiKeyVersionsSource(apiKey = ApiKeys.CONSUMER_GROUP_HEARTBEAT)
+    public void testSendingLeaveGroupHeartbeatWhenPreviousOneInFlight(final short version) {
+        mockStableMember();
+        time.sleep(DEFAULT_HEARTBEAT_INTERVAL_MS);
+        NetworkClientDelegate.PollResult result = heartbeatRequestManager.poll(time.milliseconds());
+        assertEquals(1, result.unsentRequests.size());
+        result = heartbeatRequestManager.poll(time.milliseconds());
+        assertEquals(0, result.unsentRequests.size(), "No heartbeat should be sent while a previous one is in-flight");
+
+        membershipManager.leaveGroup();
+        assertTrue(membershipManager.isLeavingGroup());
```

Comment:
nit: I don't think this line brings value here. We have ln 769, which is the end result we're interested in asserting to make sure we're actually leaving the group (and this is in the end logic belonging to the other component to achieve that result)

Reply:
This makes sense. In the unit test for 'HeartBratRequestManager,' we should focus solely on its functionality.
I will remove this line.